### DISCUSSION TITLE
Access Multidimension matrix

### DISCUSSION THREAD

**Discussion by fiji3119** [BlueBrain/HighFive#564] [Questions] [answered]
Hi,
I am new to H5 and came across this repo. I was given a file containing lookup table that is in H5 format. Using h5dump, I found that it has 4 DATASETs, but the one I am interested in has: DATASPACE  SIMPLE { ( 2, 2, 3, 2, 7, 5, 5, 7 ) / ( 2, 2, 3, 2, 7, 5, 5, 7 ) }.  Unfortunately, I am getting doxygen errors - which is not helping to learn more about HighFive APIs :( .  Are there APIs that gives DATASPACE information, like size of data, matrix dimensions etc? Is there example to open file and readonly multi-dimension data, like this one?  
I am using v2.4.0.
Thank you,
Nimesh

**Comment by 1uc** [comment, marked as answer]
You can find all examples here:
https://github.com/BlueBrain/HighFive/tree/master/src/examples

In particular, one of these may be useful:
  -  [read_write_vector_dataset.cpp](https://github.com/BlueBrain/HighFive/blob/master/src/examples/read_write_vector_dataset.cpp)
  -  [read_write_vector_dataset_references.cpp](https://github.com/BlueBrain/HighFive/blob/master/src/examples/read_write_vector_dataset_references.cpp)

If you're using boost multi-arrays, there's also an example for those. You can find API documentation here:
[bluebrain.github.io/highfive/](https://bluebrain.github.io/HighFive/)

**Reply by fiji3119** [reply]
Thank you for prompt response!!!
read_write_vector_dataset.cpp and boost_multi_array_2D.cpp have what I was looking for.
